x-pack/plugins/security_solution/public/detections/components/alerts_table/alerts_grouping.tsx

```diff
@@ -125,153 +89,146 @@ export const GroupedAlertsTableComponent: React.FC<AlertsTableComponentProps> =
 
-  useEffect(() => {
-    dispatch(updateSelectedGroup({ selectedGroup }));
-  }, [dispatch, selectedGroup]);
-
-  useInvalidFilterQuery({
-    id: tableId,
-    filterQuery: getGlobalQuery([])?.filterQuery,
-    kqlError: getGlobalQuery([])?.kqlError,
-    query: globalQuery,
-    startDate: from,
-    endDate: to,
-  });
+  const groupSelectorInRedux = useSelector((state: State) => getGroupSelector(state));
+  const selectorOptions = useRef<GroupOption[]>([]);
 
-  const { deleteQuery, setQuery } = useGlobalTime(false);
-  const additionalFilters = useMemo(() => {
-    resetPagination();
-    try {
-      return [
-        buildEsQuery(undefined, globalQuery != null ? [globalQuery] : [], [
-          ...(globalFilters?.filter((f) => f.meta.disabled === false) ?? []),
-          ...(defaultFilters ?? []),
-        ]),
-      ];
-    } catch (e) {
-      return [];
+  useEffect(() => {
+    if (
+      isNoneGroup(selectedGroups) &&
+      groupSelector.props.options.length > 0 &&
+      (groupSelectorInRedux == null ||
+        !isEqual(selectorOptions.current, groupSelector.props.options))
+    ) {
+      selectorOptions.current = groupSelector.props.options;
+      dispatch(updateGroupSelector({ groupSelector }));
+    } else if (!isNoneGroup(selectedGroups) && groupSelectorInRedux !== null) {
+      dispatch(updateGroupSelector({ groupSelector: null }));
     }
-  }, [defaultFilters, globalFilters, globalQuery, resetPagination]);
+  }, [dispatch, groupSelector, groupSelectorInRedux, selectedGroups]);
 
-  const queryGroups = useMemo(
-    () =>
-      getAlertsGroupingQuery({
-        additionalFilters,
-        selectedGroup,
-        from,
-        runtimeMappings,
-        to,
-        pageSize: pagination.pageSize,
-        pageIndex: pagination.pageIndex,
-      }),
-    [
-      additionalFilters,
-      selectedGroup,
-      from,
-      runtimeMappings,
-      to,
-      pagination.pageSize,
-      pagination.pageIndex,
-    ]
+  const [pageIndex, setPageIndex] = useState<number[]>(
+    Array(MAX_GROUPING_LEVELS).fill(DEFAULT_PAGE_INDEX)
   );
+  const [pageSize, setPageSize] = useState<number[]>(getStoragePageSize);
 
-  const {
-    data: alertsGroupsData,
-    loading: isLoadingGroups,
-    refetch,
-    request,
-    response,
-    setQuery: setAlertsQuery,
-  } = useQueryAlerts<
-    {},
-    GroupingAggregation<AlertsGroupingAggregation> &
-      GroupingFieldTotalAggregation<AlertsGroupingAggregation>
-  >({
-    query: queryGroups,
-    indexName: signalIndexName,
-    queryName: ALERTS_QUERY_NAMES.ALERTS_GROUPING,
-    skip: isNoneGroup(selectedGroup),
-  });
+  const resetAllPagination = useCallback(() => {
+    setPageIndex((curr) => curr.map(() => DEFAULT_PAGE_INDEX));
+  }, []);
 
   useEffect(() => {
-    if (!isNoneGroup(selectedGroup)) {
-      setAlertsQuery(queryGroups);
-    }
-  }, [queryGroups, selectedGroup, setAlertsQuery]);
+    resetAllPagination();
```

Comment:
What's the difference between this reset and the one on line 174? Is there a path for combining them so reset happens in one place?

Reply:
This triggers a reset when the `selectedGroup` is changed. The call on 174 resets when the global filters change. The global filters come from security solution and the selected group comes from grouping package. I will combine them in the next PR